lib/karafka/fetcher.rb

```diff
@@ -41,29 +20,17 @@ def consume_with(method_name)
       raise e
     end
 
-    # @return [Array<Karafka::Connection::ActorCluster>] array with all the connection clusters
-    # @note We cache actor clusters because in a consumer mode (without a server process), we have
-    #   to use the same clusters with the same connections in order to consume from
-    #   the same partitions
-    def actor_clusters
-      @actor_clusters ||= App
-                          .routes
-                          .each_slice(slice_size)
-                          .map do |chunk|
-                            Karafka::Connection::ActorCluster.new(chunk)
-                          end
-    end
+    private
 
-    # @return [Integer] number of topics that we want to listen to per thread
-    # @note For smaller amount of controllers it would be the best to number of controllers
-    #   to match number of threads - that way it could consume messages in "real" time
-    def slice_size
-      size = App.routes.size / Karafka::App.config.max_concurrency
-      size < 1 ? 1 : size
+    # @return [Array<Karafka::Connection::Listener>] listeners that will consume messages
+    def listeners
+      @listeners ||= App.routes.each.map do |route|
```

Comment:
Can't we use `App.routes.map` here? Seems that this `each` was needed only as a `each_slice`.